library/src/main/scala/scala/scalajs/js/annotation/JSGlobal.scala

```diff
@@ -0,0 +1,46 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js API               **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2017, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+package scala.scalajs.js.annotation
+
+/** Marks the annotated class or object as being a member of the JavaScript
+ *  global scope.
+ *
+ *  The annotated class/object must also be annotated with `@js.native`, and
+ *  therefore extend [[scala.scalajs.js.Any js.Any]].
+ *
+ *  Given:
+ *  {{{
+ *  @js.native
+ *  @JSGlobal
+ *  class Foo extends js.Object
+ *
+ *  @js.native
+ *  @JSGlobal("Foobar")
+ *  object Bar extends js.Object
+ *
+ *  @js.native
+ *  @JSGlobal("Lib.Babar")
+ *  class Babar extends js.Object
+ *  }}}
+ *
+ *  The following mappings apply (`global` denotes the global scope):
+ *
+ *  {{{
+ *  Scala.js                | JavaScript
+ *  ------------------------+------------------
+ *  new Foo()               | new global.Foo()
+ *  Bar                     | global.Foobar
+ *  js.constructorOf[Babar] | global.Lib.Babar
+ *  }}}
+ *
+ *  @see [[http://www.scala-js.org/doc/calling-javascript.html Calling JavaScript from Scala.js]]
+ */
+class JSGlobal extends scala.annotation.StaticAnnotation {
+  def this(name: String) = this()
```

Comment:
Yes, the idea is that `@JSExportTopLevel` takes over here, and since it doesn't have a default, it's all right.

I do not expect to rename `@JSExportTopLevel` to `@JSExport`. Currently `@JSExportTopLevel` is consistent between top-level classes/objects and static methods and fields. If we rename `@JSExportTopLevel` to `@JSExport`, we break this consistency, plus we reintroduce the *in*consistency between `@JSExport` on a top-level class/object versus on a member of a Scala class. If exports of instance members had been disallowed, the story would have been different.